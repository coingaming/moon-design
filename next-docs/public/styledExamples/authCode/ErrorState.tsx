import React from 'react';
import { AuthCode } from '@heathmont/moon-core';

const Example = () => (
  <AuthCode
    onChange={(value: string) => {
      console.log('New value - ', value);
    }}
    onSubmit={(value: string) => {
      console.log('Submit value - ', value);
    }}
    errorMessage="Provided value is not valid!"
  />
);

export default Example;
