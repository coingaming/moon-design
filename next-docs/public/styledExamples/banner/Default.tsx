import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
