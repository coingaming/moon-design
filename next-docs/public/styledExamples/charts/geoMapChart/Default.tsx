// @ts-nocheck
import React from 'react';
import { Filter, GeoMapChart } from '@heathmont/moon-charts';
import { SingleItemSelect } from '@heathmont/moon-components';

const Example = () => {
  const data = [
    { label: 'Estonia', code: 'EST', value: 120 },
    { label: 'Russia', code: 'RUS', value: 100 },
    { label: 'Belarus', code: 'BLR', value: 80 },
    { label: 'Argentina', code: 'ARG', value: 60 },
    { label: 'USA', code: 'USA', value: 40 },
  ];

  const filter = (
    <Filter title="by GGR">
      {() => (
        <SingleItemSelect
          onChange={() => {}}
          value="ggr"
          options={[
            { label: 'Active players', value: 'actives' },
            { label: 'GGR', value: 'ggr' },
            { label: 'Bets', value: 'bets' },
          ]}
        />
      )}
    </Filter>
  );

  return (
    <GeoMapChart
      title="Top Countries"
      data={data}
      filter={filter}
      hasUpdates={true}
      onUpdate={() => {}}
      onShare={() => {}}
      onExpand={() => {}}
    />
  );
};

export default Example;
