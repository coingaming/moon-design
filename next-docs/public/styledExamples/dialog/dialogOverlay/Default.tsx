import React from 'react';
import { DialogOverlay } from '@heathmont/moon-components';
import { Button } from '@heathmont/moon-core';

const Example = () => {
  const [showDialog, setShowDialog] = React.useState(false);

  return (
    <>
      <Button variant="tertiary" onClick={() => setShowDialog(true)}>
        Open Dialog
      </Button>
      <DialogOverlay
        isOpen={showDialog}
        onDismiss={() => setShowDialog(false)}
      />
    </>
  );
};

export default Example;
