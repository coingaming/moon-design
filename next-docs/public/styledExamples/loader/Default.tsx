import React from 'react';
import { Loader } from '@heathmont/moon-core';

const Example = () => <Loader />;

export default Example;
