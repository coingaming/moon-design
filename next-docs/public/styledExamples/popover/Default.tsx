import React, { useCallback, useEffect, useState } from 'react';
import { Popover } from '@heathmont/moon-popover';
import { Footer, Heading, Text, Button } from '@heathmont/moon-core';

const Example = () => {
  const [open, setOpen] = useState(false);

  const HandlerOpen = useCallback(
    (e: any) => {
      e.preventDefault();
      setOpen(false);
    },
    [open]
  );

  return (
    <Popover
      padding={0}
      placement="bottom"
      isOpen={open}
      onVisibilityChange={(status) => {
        setOpen(status);
        console.log('status: ', status);
      }}
      content={
        <div>
          <div className="p-8 space-y-4">
            <Heading size={20}>Agents</Heading>
            <Text size={14}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris
              nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
              reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur.
            </Text>
          </div>

          <Footer
            primButton={
              <Button fullWidth size="small" onClick={HandlerOpen}>
                OK
              </Button>
            }
            secButton={
              <Button
                fullWidth
                variant="secondary"
                size="small"
                onClick={HandlerOpen}
              >
                Cansel
              </Button>
            }
            tertButton={
              <Button
                fullWidth
                variant="tertiary"
                size="small"
                onClick={HandlerOpen}
              >
                Decline
              </Button>
            }
            isDivider
            size="small"
          />
        </div>
      }
    >
      <Button>Toggle Popover</Button>
    </Popover>
  );
};

export default Example;
