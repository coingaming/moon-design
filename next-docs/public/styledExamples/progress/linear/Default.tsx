import React from 'react';
import { ProgressLinear } from '@heathmont/moon-components';

const Example = () => (
  <>
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
    />
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
      height={5}
    />
  </>
);

export default Example;
