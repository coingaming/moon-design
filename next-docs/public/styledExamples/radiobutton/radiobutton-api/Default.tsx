import React from 'react';
import {
  RadioButtonLabel,
  RadioButtonInput,
  ListItem,
} from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col">
    <h4 className="mb-2">Please select your age:</h4>
    <div className="mb-2">
      <RadioButtonLabel id="age1">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="0 - 30" id="age1" />
          }
        >
          0 - 30
        </ListItem>
      </RadioButtonLabel>
    </div>
    <div className="mb-2">
      <RadioButtonLabel id="age2">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="0 - 30" id="age2" />
          }
        >
          31 - 60
        </ListItem>
      </RadioButtonLabel>
    </div>
    <div className="mb-2">
      <RadioButtonLabel id="age3">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="61 - 100" id="age3" />
          }
        >
          31 - 60
        </ListItem>
      </RadioButtonLabel>
    </div>
  </div>
);

export default Example;
