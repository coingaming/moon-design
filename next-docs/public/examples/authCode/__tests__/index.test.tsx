/**
 * @jest-environment jsdom
 */

import { cleanup, render } from '@testing-library/react';
import Default from '../Default';
import FourChars from '../FourChars';
import OnlyDigits from '../OnlyDigits';
import ErrorState from '../ErrorState';

const withRtl = (component: JSX.Element) => <div dir="rtl">{component}</div>;

describe('AuthCode', () => {
  afterEach(cleanup);
  it('renders Default', () => {
    const tree = render(<Default />);
    expect(tree).toMatchSnapshot();
  });
  it('renders FourChars', () => {
    const tree = render(<FourChars />);
    expect(tree).toMatchSnapshot();
  });
  it('renders OnlyDigits', () => {
    const tree = render(<OnlyDigits />);
    expect(tree).toMatchSnapshot();
  });
  it('renders ErrorState', () => {
    const tree = render(<ErrorState />);
    expect(tree).toMatchSnapshot();
  });
});

describe('AuthCode in RTL', () => {
  afterEach(cleanup);
  it('renders Default', () => {
    const tree = render(withRtl(<Default />));
    expect(tree).toMatchSnapshot();
  });
  it('renders FourChars', () => {
    const tree = render(withRtl(<FourChars />));
    expect(tree).toMatchSnapshot();
  });
  it('renders OnlyDigits', () => {
    const tree = render(withRtl(<OnlyDigits />));
    expect(tree).toMatchSnapshot();
  });
  it('renders ErrorState', () => {
    const tree = render(withRtl(<ErrorState />));
    expect(tree).toMatchSnapshot();
  });
});
