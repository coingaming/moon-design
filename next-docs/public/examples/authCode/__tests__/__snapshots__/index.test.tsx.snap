// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthCode renders 1`] = `
<div
  className="sc-fEVxLL iLfGGv"
  dir="ltr"
>
  <div />
</div>
`;

exports[`AuthCode renders Custom length 1`] = `
<div
  className="sc-fEVxLL iLfGGv"
  dir="ltr"
>
  <div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX dtoFCR"
        disabled={false}
      >
        <input
          className="sc-goMRkL JOvHL"
          dir="ltr"
          disabled={false}
          id="auth-code-part-0"
          inputMode="text"
          maxLength={4}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL JOvHL"
          dir="ltr"
          disabled={true}
          id="auth-code-part-1"
          inputMode="text"
          maxLength={4}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL JOvHL"
          dir="ltr"
          disabled={true}
          id="auth-code-part-2"
          inputMode="text"
          maxLength={4}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL JOvHL"
          dir="ltr"
          disabled={true}
          id="auth-code-part-3"
          inputMode="text"
          maxLength={4}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AuthCode renders Error state 1`] = `
<div
  className="sc-fEVxLL gTsxvH"
  dir="ltr"
>
  <div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX dtoFCR"
        disabled={false}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={false}
          id="auth-code-part-0"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={true}
          id="auth-code-part-1"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={true}
          id="auth-code-part-2"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={true}
          id="auth-code-part-3"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={true}
          id="auth-code-part-4"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-jdzWJC hBoQHx"
    >
      <div
        className="sc-iOnGvX btMRNz"
        disabled={true}
      >
        <input
          className="sc-goMRkL kDWyPF"
          dir="ltr"
          disabled={true}
          id="auth-code-part-5"
          inputMode="text"
          maxLength={6}
          onChange={[Function]}
          pattern="[a-z0-9]*"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-fCdBJp fFJVf"
    >
      <p
        className="sc-cQIpJi jgOjbU"
        color="chiChi.100"
        size={12}
      >
        Provided value is not valid!
      </p>
    </div>
  </div>
</div>
`;

exports[`AuthCode renders Only digits 1`] = `
<div
  className="sc-fEVxLL iLfGGv"
  dir="ltr"
>
  <div />
</div>
`;
